library/Network/IPFS/Pin.hs

```diff
@@ -10,7 +10,16 @@ import qualified Network.IPFS.Internal.UTF8       as UTF8
 import qualified Network.IPFS.Client.Pin     as Pin
 import           Network.IPFS.Add.Error      as IPFS.Add
 import           Network.IPFS.Types          as IPFS
+import           Servant.Client.Core
 
+-- | TODO: Move to a file for IPFS client types
+data IPFSErrorBody = IPFSErrorBody {message :: String}
```

Comment:
@expede Question for you, where do you think would be the best place to locate a data type like this?

Reply:
Good question!

Types should live in their own module. By convention, the type `Foo` would live in `MyApp.Some.Path.Foo.Types`

In this case, I'd move `data IPFSErrorBody`, and the `FromJSON` instance to `Network.IPFS.Pin.ErrorBody.Types`, and drop the `IPFS` from the front of the name (because we can always import it `qualified ... as IPFS`), and it's implied that we're working in IPFS because we're under `Network.IPFS`.

Reply:
Ok, good explanation. I think with this in mind Ill be putting it under `Network.IPFS.Client.Error.types` since its more a generic error shape returned by calls to IPFS